estate/models/estate_property.py

```diff
@@ -43,4 +46,78 @@ class EstateProperty(models.Model):
         copy=False,
         default="new",
     )
+    active = fields.Boolean("Active", default=True)
+
+    
     property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    user_id = fields.Many2one("res.users", string="User", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+
+    total_area = fields.Integer(
+        "Total Area ",
+        compute="_compute_total_area",
+        help="Total area  = the living area + the garden area",
+    )
+
+
+    best_offer = fields.Float("Best Offer", compute="_compute_best_offer", help="Best offer received")
+
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+
+
+    @api.depends("offer_ids.price")
+    def _compute_best_offer(self):
+        for prop in self:
+            prop.best_price = max(prop.offer_ids.mapped("price")) if prop.offer_ids else 0.0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+         for estate in self:
+            if not estate.garden:
+                 estate.garden_area=0
+
+
+    
+    @api.onchange("date_availability")
+    def _onchange_date_availability(self):
+        for estate in self:
+           return {
+         "warning": {
+             "title": "Warning",
+             "message": "What is this?"
+                        }
+            }
+
+
+
+    @api.constrains("expected_price", "selling_price")
+    def _check_price_difference(self):
+        for prop in self:
+            if (
+                not float_is_zero(prop.selling_price, precision_rounding=0.01)
+                and float_compare(prop.selling_price, prop.expected_price * 90.0 / 100.0, precision_rounding=0.01) < 0
+            ):
+                raise ValidationError(
+                    "The selling price must be at least 90% of the expected price! "
+                    + "You must reduce the expected price if you want to accept this offer."
+                )
+
+    def action_sold(self):
+        if "canceled" in self.mapped("state"):
+            raise UserError("Canceled properties cannot be sold.")
+        return self.write({"state": "sold"})
```

Comment:
Le recordset ne contient qu'un record unique lorsqu'on clique un boutton dans une form view
```suggestion
    def action_sold(self):
        self.ensure_one()
        if "canceled" == self.state:
            raise UserError("Canceled properties cannot be sold.")
        return self.write({"state": "sold"})
```

Reply:
Ah oui merci!